pkg/util/bandwidth/utils.go

```diff
@@ -36,31 +36,53 @@ func validateBandwidthIsReasonable(rsrc *resource.Quantity) error {
 }
 
 // ExtractPodBandwidthResources extracts the ingress and egress from the given pod annotations
-func ExtractPodBandwidthResources(podAnnotations map[string]string) (ingress, egress *resource.Quantity, err error) {
+func ExtractPodBandwidthResources(podAnnotations map[string]string) (ingress, ingressBurst, egress, egressBurst *resource.Quantity, err error) {
 	if podAnnotations == nil {
-		return nil, nil, nil
+		return nil, nil, nil, nil, nil
 	}
 	str, found := podAnnotations["kubernetes.io/ingress-bandwidth"]
 	if found {
 		ingressValue, err := resource.ParseQuantity(str)
 		if err != nil {
-			return nil, nil, err
+			return nil, nil, nil, nil, err
 		}
 		ingress = &ingressValue
 		if err := validateBandwidthIsReasonable(ingress); err != nil {
-			return nil, nil, err
+			return nil, nil, nil, nil, err
+		}
+	}
+	str, found = podAnnotations["kubernetes.io/ingress-burst-bandwidth"]
+	if found {
+		ingressBurstValue, err := resource.ParseQuantity(str)
+		if err != nil {
+			return nil, nil, nil, nil, err
+		}
+		ingressBurst = &ingressBurstValue
+		if err := validateBandwidthIsReasonable(ingressBurst); err != nil {
```

Comment:
Do we need some additional validation here for the burst rate? e.g. from the [tc docs](http://man7.org/linux/man-pages/man8/tc-tbf.8.html) and we could validate that these values are reasonable.

```
If your buffer is too small, packets may be dropped because
more tokens arrive per timer tick than fit in your bucket.
The minimum buffer size can be calculated by dividing the rate
by HZ.
```

Reply:
We can make sure `ingressBurst` always greater `ingressRate`.